mailscanner/database.php

```diff
@@ -47,7 +47,7 @@ private function __construct()
      */
     public static function connect($host = '', $username = '', $password = '', $database = '')
     {
-        if (self::$link == null || !self::$link) {
+        if (self::$link instanceof mysqli) {
```

Comment:
this should be negative:
`if (!self::$link instanceof mysqli) {`
`instanceof` has precedence over `!` as operator